packages/lib/fs-driver-base.ts

```diff
@@ -70,7 +70,11 @@ export default class FsDriverBase {
 			// Check if the filename does not exist in the filesystem and is not reserved
 			const exists = await this.exists(nameToTry) || reservedNames.includes(nameToTry);
 			if (!exists) return nameToTry;
-			nameToTry = `${nameNoExt} (${counter})${extension}`;
+			if (!markdownSafe) {
+				nameToTry = `${nameNoExt} (${counter})${extension}`;
+			} else {
+				nameToTry = `${nameNoExt}-${counter}${extension}`;
+			}
```

Comment:
The default friendlySafeFilename format does not produce valid markdown links, rather than escaping the markdown links (which is ugly for users), I decided to add an option to generate filenames that are valid in markdown.